data/cognates/Collecting_cognates_dataset.ipynb

```diff
@@ -0,0 +1,489 @@
+       "      <th>first_word</th>\n",
+       "      <th>second_word</th>\n",
+       "    </tr>\n",
+       "  </thead>\n",
+       "  <tbody>\n",
+       "    <tr>\n",
+       "      <th>0</th>\n",
+       "      <td>А</td>\n",
+       "      <td>А-конто</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>1</th>\n",
+       "      <td>А</td>\n",
+       "      <td>Аканье</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>2</th>\n",
+       "      <td>А</td>\n",
+       "      <td>Акать</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>3</th>\n",
+       "      <td>А</td>\n",
+       "      <td>Акающий</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>4</th>\n",
+       "      <td>А</td>\n",
+       "      <td>Ашка</td>\n",
+       "    </tr>\n",
+       "  </tbody>\n",
+       "</table>\n",
+       "</div>"
+      ],
+      "text/plain": [
+       "  first_word second_word\n",
+       "0          А     А-конто\n",
+       "1          А      Аканье\n",
+       "2          А       Акать\n",
+       "3          А     Акающий\n",
+       "4          А        Ашка"
+      ]
+     },
+     "execution_count": 6,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "import pandas as pd\n",
+    "\n",
+    "data = pd.read_csv('cognates_full.csv', encoding='utf_8')\n",
+    "data.head()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def dist_Levenshtein(s1, s2):\n",
+    "    d = [[0] * (len(s2) + 1) for i in range(len(s1) + 1)]\n",
+    "    for i in range(len(s1) + 1):\n",
+    "        for j in range(len(s2) + 1):\n",
+    "            if i == 0:\n",
+    "                d[i][j] = j\n",
+    "            elif j == 0:\n",
+    "                d[i][j] = i\n",
+    "            else:\n",
+    "                d[i][j] = min(d[i][j - 1] + 1, d[i - 1][j] + 1)\n",
+    "                delta = 0 if s1[i - 1] == s2[j - 1] else 1\n",
+    "                d[i][j] = min(d[i][j], d[i - 1][j - 1] + delta)\n",
+    "                \n",
+    "    return d[len(s1)][len(s2)]"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "4\n"
+     ]
+    }
+   ],
+   "source": [
+    "print(dist_Levenshtein('Levenshtein', 'Lenvinsten'))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def longest_common_substring(s1, s2):\n",
+    "    d = [[0] * (len(s2) + 1) for i in range(len(s1) + 1)]\n",
+    "    for i in range(len(s1) + 1):\n",
+    "        for j in range(len(s2) + 1):\n",
+    "            if i and j and s1[i - 1] == s2[j - 1]:\n",
+    "                d[i][j] = d[i - 1][j - 1] + 1\n",
+    "    max_val = 0\n",
```

Comment:
Let's search `max_val` during the first loop, when we fill our dynamic array. :)